src/state/models/mymodel.ts

```diff
@@ -0,0 +1,39 @@
+import { createModel } from '@captaincodeman/rdx-model'
+import { State } from '../store'
+import { createSelector } from 'reselect'
+
+export interface MyModelState {
+  input: Number
+  squaredValue: String
+}
+
+// Please note that this default export of createModel() is re-exported 
+// as mymodel in ../ index.ts, then consumed by getState() below 
+// as state.mymodel.
+// This could be confusing, even if it does make perfect sense, eventually
+export default createModel({
+  
+  state: <MyModelState>{
+    input: 1,
+    squaredValue: "Input value has not been set"
+  },
+
+  reducers: {
+    takeValue(state, input) {
+      const result =  input * input
+      const returning: String = `${input} times itself is ${result} at ${Date.now()}`
+      return { ...state, squaredValue: returning }
+    }
+  }
+})
+
+// see ../index.ts for where state.mymodel below,
+// is defined using default export from above
+const getState = (state: State) => state.mymodel
+
+export namespace MyModelSelectors {
```

Comment:
For a simplest / getting started / learning concepts I would leave the selectors out of it - it's an optimization and provides some benefits, but isn't necessary to learn / use (esp as, in this case, it isn't adding anything as it's just mapping a simple value)